### ISSUE TITLE
Flutter-web: Snackbar not read in NVDA screen reader

### ISSUE THREAD

**Issue by whsliuyun** [flutter/flutter#108768]
## Steps to Reproduce

1. Window's NVDA screen reader on.
2. Run flutter web with example snackbar app. https://docs.flutter.dev/cookbook/design/snackbars#interactive-example
3. Click button to show snackbar.

**Expected results:** Snackbar content should be read

**Actual results:** It is not read.

I did some investigation, per MDN
Simply including an aria-live attribute or a specialized live region role (such as role="alert") in the initial markup as it's loaded will have no effect.
Dynamically adding an element with an aria-live attribute or specialized role to the document also won't result in any announcement by assistive technologies (as, at that point, the browser/assistive technologies are not aware of the live region yet, so cannot monitor it for changes).
Always make sure that the live region is present in the document first, and only then dynamically add/change any content.

So we should first have an empty aria-live within dom and then change the content. However I don't see any workaround for this.

Semantic(liveregion:true, child: Text('')) won't create a live region since the content is empty. I couldn't add a meaningful live-region element either, because other platform: native Android works fine.

Any suggestions?

**Comment by yjbanov** [comment]
A little more details:

* This shows up in JAWS and NVDA on Windows.
* This sometimes happens in Chrome on Android.

A few ideas:

* Delay the content of live regions in the Flutter Web implementation: the idea is to "trick" the AT into reading the contents of the live region.
* Use global persistent live regions and copy the contents into them. This would satisfy the requirement that a "live region is present in the document first". This could get messy though because we'd need to maintain the same content in multiple places.

Also, now that we know that live regions on the web have the "live region is present in the document first" requirement, we could also allow creating empty live regions. This would allow the developer to control when the content appears.

**Comment by whsliuyun** [comment]
I'd prefer not having developers create a new empty live regions, this add developer work.
I like the idea of delay.
1. When the semantic dom needs to be updated, we need to check if we need to insert a new aria-live element.
2. If a new element needs to be ADDED, flutter system should add a delay between (insert empty aria-live element to dom) and (update the content).
3. For update already existing aria-live element content, no delay is required.

This should benefit for all Semantic liveregion use case automatically for developers, not just for snackbar.

**Comment by huycozy** [comment]
Hi @whsliuyun
This issue is reproducible on the latest `stable` and `master` channels.

- Chrome (Windows): ❌
- Chrome (Android): ✅

✅: No Issue ❌: Issue reproduced

<details>
<summary>Demo</summary>

| Windows | Android |
| --------------- | --------------- |
<video src="https://user-images.githubusercontent.com/104349824/182334813-a13a7f67-a90f-49ab-a57f-ad8e43031b14.mp4"/> | <video src="https://user-images.githubusercontent.com/104349824/182334799-be60446e-d882-4562-9b8a-01ec646fc701.mp4"/>

</details>

<details>
<summary>flutter doctor -v</summary>

```bash
[√] Flutter (Channel stable, 3.0.5, on Microsoft Windows [Version 10.0.19043.1766], locale en-US)
    • Flutter version 3.0.5 at C:\WIP\flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision f1875d570e (9 hours ago), 2022-07-13 11:24:16 -0700
    • Engine revision e85ea0e79c
    • Dart version 2.17.6
    • DevTools version 2.12.2

[√] Android toolchain - develop for Android devices (Android SDK version 32.0.0)
    • Android SDK at C:\AndroidSDK
    • Platform android-32, build-tools 32.0.0
    • ANDROID_HOME = C:\AndroidSDK
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 11.0.12+7-b1504.28-7817840)
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files (x86)\Google\Chrome\Application\chrome.exe

[√] Visual Studio - develop for Windows (Visual Studio Community 2022 17.2.5)
    • Visual Studio at C:\Program Files\Microsoft Visual Studio\2022\Community
    • Visual Studio Community 2022 version 17.2.32616.157
    • Windows 10 SDK version 10.0.19041.0

[√] Android Studio (version 2021.2)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.12+7-b1504.28-7817840)

[!] Android Studio (version 2022.1)
    • Android Studio at C:\Program Files\Android\Android Studio Electric Eel 2022.1.1
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    X Unable to find bundled Java version.
    • Try updating or re-installing Android Studio.

[√] VS Code (version 1.69.1)
    • VS Code at C:\Users\ADMIN\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.44.0

[√] Connected device (3 available)
    • Windows (desktop) • windows • windows-x64    • Microsoft Windows [Version 10.0.19043.1766]
    • Chrome (web)      • chrome  • web-javascript • Google Chrome 103.0.5060.114
    • Edge (web)        • edge    • web-javascript • Microsoft Edge 101.0.1210.53

[√] HTTP Host Availability
    • All required HTTP hosts are available

! Doctor found issues in 1 category.
```

```bash
[√] Flutter (Channel master, 3.1.0-0.0.pre.2030, on Microsoft Windows [Version 10.0.19043.1826], locale en-US)
    • Flutter version 3.1.0-0.0.pre.2030 on channel master at C:\WIP\flutter_master
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision dcdb9f1cde (88 minutes ago), 2022-08-02 03:21:07 -0400
    • Engine revision 3b2bd24a1e
    • Dart version 2.19.0 (build 2.19.0-53.0.dev)
    • DevTools version 2.16.0

[√] Android toolchain - develop for Android devices (Android SDK version 32.0.0)
    • Android SDK at C:\AndroidSDK
    • Platform android-32, build-tools 32.0.0
    • ANDROID_HOME = C:\AndroidSDK
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 11.0.12+7-b1504.28-7817840)
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files (x86)\Google\Chrome\Application\chrome.exe

[√] Visual Studio - develop for Windows (Visual Studio Community 2022 17.2.5)
    • Visual Studio at C:\Program Files\Microsoft Visual Studio\2022\Community
    • Visual Studio Community 2022 version 17.2.32616.157
    • Windows 10 SDK version 10.0.19041.0

[√] Android Studio (version 2021.2)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.12+7-b1504.28-7817840)

[!] Android Studio (version 2022.1)
    • Android Studio at C:\Program Files\Android\Android Studio Electric Eel 2022.1.1
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    X Unable to find bundled Java version.
    • Try updating or re-installing Android Studio.

[√] VS Code (version 1.69.2)
    • VS Code at C:\Users\ADMIN\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.44.0

[√] Connected device (3 available)
    • Windows (desktop) • windows • windows-x64    • Microsoft Windows [Version 10.0.19043.1826]
    • Chrome (web)      • chrome  • web-javascript • Google Chrome 103.0.5060.134
    • Edge (web)        • edge    • web-javascript • Microsoft Edge 101.0.1210.53

[√] HTTP Host Availability
    • All required HTTP hosts are available

! Doctor found issues in 1 category.
```
</details>

But I see there's an open issue #75880 (flutter app is accessible when uses NVDA) that hasn't been fixed yet. Do you see it as the meta-issue of this issue?

**Comment by yjbanov** [comment]
I'd prefer to keep this issue. It's more actionable. I'll see if we should close #75880. I'm not sure what it means to fix that. It's probably a matter of fixing many little bugs.
